Makefile

```diff
@@ -1,174 +1,171 @@
+PACKAGE_NAME := github.com/fabric8-services/${PROJECT_NAME}
 SOURCE_DIR ?= .
 SOURCES := $(shell find $(SOURCE_DIR) -type d \( -name vendor -o -name .glide \) -prune -o -name '*.go' -print)
 VENDOR_DIR=vendor
 LDFLAGS := -w
-BINARY := fabric8-toggles-service
 DESIGN_DIR=design
 DESIGNS := $(shell find $(SOURCE_DIR)/$(DESIGN_DIR) -path $(SOURCE_DIR)/vendor -prune -o -name '*.go' -print)
 GOAGEN_BIN=$(VENDOR_DIR)/github.com/goadesign/goa/goagen/goagen
 CUR_DIR=$(shell pwd)
-INSTALL_PREFIX=$(CUR_DIR)/bin
-DOCKER_BIN_NAME=docker
-DOCKER_BIN := $(shell command -v $(DOCKER_BIN_NAME) 2> /dev/null)
-GLIDE_BIN_NAME := glide
-GLIDE_BIN := $(shell command -v $(GLIDE_BIN_NAME) 2> /dev/null)
+BUILD_DIR = bin
+
 # This pattern excludes some folders from the coverage calculation (see grep -v)
 ALL_PKGS_EXCLUDE_PATTERN = 'vendor\|app\|tool\/cli\|design\|client\|test'
 
 
-COMMIT=$(shell git rev-parse HEAD)
+IMAGE_TAG ?= $(shell git rev-parse HEAD)
 GITUNTRACKEDCHANGES := $(shell git status --porcelain --untracked-files=no)
 ifneq ($(GITUNTRACKEDCHANGES),)
-COMMIT := $(COMMIT)-dirty
+IMAGE_TAG := $(IMAGE_TAG)-dirty
 endif
 BUILD_TIME=`date -u '+%Y-%m-%dT%H:%M:%SZ'`
+# Pass in build time variables to main
+LDFLAGS=-ldflags "-X ${PACKAGE_NAME}/controller.Commit=${IMAGE_TAG} -X ${PACKAGE_NAME}/controller.BuildTime=${BUILD_TIME}"
+
+.DEFAULT_GOAL := help
+
+GOANALYSIS_DIRS=$(shell go list -f {{.Dir}} ./... | grep -vEf .goanalysisignore)
+GOANALYSIS_PKGS=$(shell go list -f {{.ImportPath}} ./... | grep -vEf .goanalysisignore)
+GOANALYSIS_FILES=$(shell find  . -name '*.go' | grep -vEf .goanalysisignore)
+# Check that given variables are set and all have non-empty values,
+# die with an error otherwise.
+#
+# Params:
+#   1. Variable name(s) to test.
+#   2. (optional) Error message to print.
+check_defined = \
+    $(strip $(foreach 1,$1, \
+        $(call __check_defined,$1,$(strip $(value 2)))))
+__check_defined = \
+    $(if $(value $1),, \
+      $(error Undefined $1$(if $2, ($2))))
+
+all: tools generate fmtcheck test image ## Compiles binary and runs format and style checks
+
+$(BUILD_DIR): 
+	mkdir $(BUILD_DIR)
 
-PACKAGE_NAME := github.com/fabric8-services/fabric8-toggles-service
+.PHONY: build
+build: vendor generate $(BUILD_DIR) # Builds the Linux binary for the container image into $BUILD_DIR
+	go build -v $(LDFLAGS) -o $(BUILD_DIR)/$(REGISTRY_IMAGE)
 
-# For the global "clean" target all targets in this variable will be executed
-CLEAN_TARGETS =
+.PHONY: build-linux
+build-linux: vendor generate $(BUILD_DIR) # Builds the Linux binary for the container image into $BUILD_DIR
+	CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -v $(LDFLAGS) -o $(BUILD_DIR)/$(REGISTRY_IMAGE)
 
-# Pass in build time variables to main
-LDFLAGS=-ldflags "-X ${PACKAGE_NAME}/controller.Commit=${COMMIT} -X ${PACKAGE_NAME}/controller.BuildTime=${BUILD_TIME}"
+image: clean-artifacts $(BUILD_DIR)/$(REGISTRY_IMAGE) ## Builds the container image using the binary compiled for Linux
```

Comment:
$(BUILD_DIR)/$(REGISTRY_IMAGE) has been rename to `build` therefore the CentOS CI error `make: *** No rule to make target `bin/fabric8-toggles-service', needed by `image'.  Stop.`